src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.ICSharpProjectSite.cs

```diff
@@ -124,9 +123,14 @@ public int GetValidStartupClasses(IntPtr[] classNames, ref int count)
             // If classNames is NULL, then we need to populate the number of valid startup
             // classes only
             var project = Workspace.CurrentSolution.GetProject(VisualStudioProject.Id);
-            var compilation = project.GetCompilationAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
+            if (project == null)
+            {
+                count = 0;
+                return VSConstants.S_OK;
```

Comment:
This wouldn't return some sort of error code in this case?

Reply:
Also, if this is called with passing in a class names array, do we need to clear out the array or something?

Reply:
This is not necessary today though there is no harm in doing so. We are first called to determine the number of elements to expect, then the native codebase allocates the array, then it calls us again to fill it.

Reply:
note that all other error cases in thie method today also simply set the count to 0. returning a count of zero is interpreted as ignoring the array values

Reply:
In testing this cannot happen.